doc/neps/roadmap.rst

```diff
@@ -1,5 +1,5 @@
 =============
-NumPy Roadmap
+NumPy roadmap
```

Comment:
I am mixed on this, Roadmap is usually capitalized everywhere I look... I don't have strong feelings either way but not sure about the sentence case rule here.

Reply:
![Screenshot 2021-09-08 at 8 24 04 PM](https://user-images.githubusercontent.com/17350312/132533793-96486523-ea8c-44b0-a9f9-90cce172c33d.png)
It's a proper noun too.